lib/string_reverse.rb

```diff
@@ -1,6 +1,25 @@
 # A method to reverse a string in place.
 # Time complexity: ?
 # Space complexity: ?
+
 def string_reverse(my_string)
-  raise NotImplementedError
+  # raise NotImplementedError
+
+  if my_string == nil
+    return my_string
+  else
+    i = 0
+    while my_string[i] != nil
+      i += 1
+    end #while end
+
+    for j in (0...(i / 2))
+      temp = ""

```

Comment:
There is no need to initialize temp here since you reassign it immediately on the next line.